### ISSUE TITLE
Avoiding permission changes when sharing volumes

### ISSUE THREAD

**Issue by ConYel** [rocker-org/rocker#358]
Hello,
I would like to run rstudio without changing the permissions on a folder sharing volume.
I run the suggested  code :
`docker run -d -P -v $(pwd):/home/$USER/foo -e USERID=$UID rocker/rstudio`
but it never starts the services:

`sudo docker run --rm -v /root/Documenti/R_virtual_shared_folder/Documents/:/home/rstudio/data -e PASSWORD=12345 -e USER=$UID -p 8787:8787 congelos/rocker_tidyverse_plus_de_pckages`

> [s6-init] making user provided files available at /var/run/s6/etc...exited 0. 
> [s6-init] ensuring user provided files have correct perms...exited 0.
> [fix-attrs.d] applying ownership & permissions fixes...
> [fix-attrs.d] done.
> [cont-init.d] executing container initialization scripts...
> [cont-init.d] add: executing... 
> Nothing additional to add
> [cont-init.d] add: exited 0.
> [cont-init.d] userconf:  executing...

my machine :
CentOS Linux release 7.6.1810 (Core) 
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

CentOS Linux release 7.6.1810 (Core) 
CentOS Linux release 7.6.1810 (Core)

**Edit: I run as root**

Thank you for your time
Constantinos

**Comment by cboettig** [comment]
Thanks, can you give us the Docker version info too?  In particular, have you tried running with the latest version of Docker installed from their site (e.g. via https://get.docker.com) or are you using the version packaged with Centos?  (I recall some permission issues with the latter which you can possibly dig up from another issue in the lists here but don't recall the details -- we're mostly a debian world on our end)

**Comment by ConYel** [comment]
> Client:
>  Debug Mode: false
> 
> Server:
>  Containers: 0
>   Running: 0
>   Paused: 0
>   Stopped: 0
>  Images: 25
>  Server Version: 19.03.2-beta1
>  Storage Driver: overlay2
>   Backing Filesystem: xfs
>   Supports d_type: true
>   Native Overlay Diff: true
>  Logging Driver: json-file
>  Cgroup Driver: cgroupfs
>  Plugins:
>   Volume: local
>   Network: bridge host ipvlan macvlan null overlay
>   Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
>  Swarm: inactive
>  Runtimes: runc
>  Default Runtime: runc
>  Init Binary: docker-init
>  containerd version: 894b81a4b802e4eb2a91d1ce216b8817763c29fb
>  runc version: 425e105d5a03fabd737a126ad93d62a9eeede87f
>  init version: fec3683
>  Security Options:
>   seccomp
>    Profile: default
>  Kernel Version: 3.10.0-957.27.2.el7.x86_64
>  Operating System: CentOS Linux 7 (Core)
>  OSType: linux
>  Architecture: x86_64
>  CPUs: 32
>  Total Memory: 251.7GiB
>  Name: localhost.localdomain
>  ID: HVJS:7PEY:ZXHK:2WCS:WVEQ:CB4P:52EW:ZZEH:TSFI:UM2Y:BW3M:G3VQ
>  Docker Root Dir: /var/lib/docker
>  Debug Mode: false
>  Username: congelos
>  Registry: https://index.docker.io/v1/
>  Labels:
>  Experimental: false
>  Insecure Registries:
>   127.0.0.0/8
>  Live Restore Enabled: false


`Docker version 19.03.2-beta1, build a0eaa12`

Is the information mentioned above enough? 

No, I haven't checked the latest version of docker, I'll search about the issues regarding Centos.

**Comment by cboettig** [comment]
your docker does look pretty recent, though I can't actually tell if that's installed directly or is the one centos is packaging in the repos?  

Did you check the docker logs?  

Note that your command:

```
docker run -d -P -v $(pwd):/home/$USER/foo -e USERID=$UID rocker/rstudio
```

is not specifying a password, which is now required.  (Also usually I prefer to do an explicit port like you do in your later example).  

Your other interactive command looks like it is working correctly from the log you show, so I'm not sure what error you are seeing.  You should then be able to connect to RStudio via the port you bound (8787) unless blocked by firewall.

**Comment by ConYel** [comment]
I'm sorry I wasn't clear enough, I didn't use the 
`docker run -d -P -v $(pwd):/home/$USER/foo -e USERID=$UID rocker/rstudio` 
command, only the second one is the one I run. The problem is that it stops in  : 
> Nothing additional to add
> [cont-init.d] add: exited 0.
> [cont-init.d] userconf: executing...
when I run :
`sudo docker run --rm -v /root/Documenti/R_virtual_shared_folder/Documents/:/home/rstudio/data -e PASSWORD=12345  -p 8787:8787 congelos/rocker_tidyverse_plus_de_pckages`
 __e.g. without user specified__ it starts the services and I can log in Rstudio throught browser correctly.

> [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
> [s6-init] ensuring user provided files have correct perms...exited 0.
> [fix-attrs.d] applying ownership & permissions fixes...
> [fix-attrs.d] done.
> [cont-init.d] executing container initialization scripts...
> [cont-init.d] add: executing... 
> Nothing additional to add
> [cont-init.d] add: exited 0.
> [cont-init.d] userconf: executing... 
> __[cont-init.d] userconf: exited 0.__
> __[cont-init.d] done.__
>__[services.d] starting services__
> [services.d] done.

**Comment by ConYel** [comment]
> 
> 
> your docker does look pretty recent, though I can't actually tell if that's installed directly or is the one centos is packaging in the repos?
> 
> Did you check the docker logs?
> 
> Note that your command:
> 
> ```
> docker run -d -P -v $(pwd):/home/$USER/foo -e USERID=$UID rocker/rstudio
> ```
> 
> is not specifying a password, which is now required. (Also usually I prefer to do an explicit port like you do in your later example).
> 
> Your other interactive command looks like it is working correctly from the log you show, so I'm not sure what error you are seeing. You should then be able to connect to RStudio via the port you bound (8787) unless blocked by firewall.

How could I find the information about how it was installed?
I believe that they followed instruction from the docker site: 
https://docs.docker.com/install/linux/docker-ce/centos/

**Comment by ConYel** [comment]
After running this command :

`docker run --rm -v $(pwd):/home/rstudio -p 8787:8787 -e PASSWORD=12345 -e USER=$UID congelos/rocker_tidyverse_plus_de_pckages`

and waiting for some hours
In the end, it started

> [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] add: executing... 
Nothing additional to add
[cont-init.d] add: exited 0.
[cont-init.d] userconf: executing... 
USER is now 0
[cont-init.d] userconf: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.

The first issue I encountered was that I couldn't use the __*rstudio*__ username anymore and I had to type __*0*__.

Something that I found  strange was that
while I was waiting for the docker  to start I found on htop a command:
`cp -r /home/rstudio /0 `

My fault was that I had put the volume linking to ` /home/rstudio`, it seems that it was copying everything from one directory (GBs of data) to the /home/0 dir. Finally, I had rerun it with linking it to the dir `/home/0` e.g. `docker run --rm -v $(pwd):/home/0 -p 8787:8787 -e PASSWORD=12345 -e USER=$UID congelos/rocker_tidyverse_plus_de_pckages`  and everything seems to work now.

Sorry for the trouble.

Kind regards
Constantinos
